webapp/javascript/util/formatDate.js

```diff
@@ -10,32 +11,65 @@ const multiplierMapping = {
   y: "year",
 };
 
-export default function humanReadableRange(until, from) {
-  if (until === "now" && typeof from === "string") {
-    const m = from.match(/^now-(?<number>\d+)(?<multiplier>\D+)$/);
-    if (m && multiplierMapping[m.groups.multiplier]) {
-      let multiplier = multiplierMapping[m.groups.multiplier];
-      if (m.groups.number > 1) {
-        multiplier += "s";
-      }
-      const readableDateForm = {
-        range: `Last ${m.groups.number} ${multiplier}`,
-        from: moment().add(-m.groups.number, multiplier).toDate(),
-        until: moment().toDate(),
+function convertPresetsToDate(from) {
+  const { groups } = from.match(/^now-(?<number>\d+)(?<multiplier>\D+)$/);
+  const { number, multiplier } = groups;
+  let _multiplier = multiplierMapping[multiplier];
+  if (number > 1) {
+    _multiplier += "s";
+  }
+  // convert preset to a Date object
+  const fromObjectForm = moment().add(-number, _multiplier).toDate();
+
+  return { fromObjectForm, number, _multiplier };
+}
+
+function formateDate(from, until, preset = true) {
+  const { fromObjectForm, number, _multiplier } = preset
```

Comment:
How does function handle case where the de-structured variables are null?

Reply:
in that case, it gets replaced by another value but that's only for `fromObjectForm` but the other two variables stay unhandled.
is there a better way to handle it?

Reply:
Think about from perspective of user. What experience would user have currently if those values were null, what would I as a user expect if something wasn't working correctly

Reply:
`number` and `_multiplier` are used only in the default range value when `/^now-/.test(from` is true and `until === "now" `. otherwise the variables won't be used in any part of the code.
so I think there is no need to handle it.

Reply:
What if I decided to use this function in a different context/different use case?